docs/sources/security.md

```diff
@@ -0,0 +1,167 @@
+
+## Beyla operation modes
+
+Beyla can operate in two distinct modes: _application observability_ and _network observability_. These modes are not mutually exclusive and can be used together as needed. The former is typically enabled using the `BEYLA_OPEN_PORT` or `BEYLA_EXECUTABLE_NAME` options, while the latter is controlled via the `BEYLA_NETWORK_METRICS` option. For more details on enabling these modes, refer to the [configuration documentation](/docs/beyla/latest/configure/options/).
+
+What Beyla requires in terms of capabilities depends entirely on the modes and features you enable. Conversely, the capabilities you provide determine what Beyla is able to do.
+
+When starting up, Beyla reads its configuration and checks for the required capabilities. If any are missing, Beyla displays a warning like the following:
+
+```
+time=2025-01-27T17:21:20.197-06:00 level=WARN msg="Required system capabilities not present, Beyla may malfunction" error="the following capabilities are required: CAP_DAC_READ_SEARCH, CAP_BPF, CAP_CHECKPOINT_RESTORE"
+```
+
+Beyla then attempts to continue running, but missing capabilities may lead to errors later on.
+
+To prevent this, you can set `BEYLA_ENFORCE_SYS_CAPS=1`, which causes Beyla to fail immediately if the required capabilities are not available. In this case, it terminates right after printing the warning message above.
+
+## List of capabilities required by Beyla
+
+Below is a list of capabilities and their usage in the context of Beyla
+
+| Capability               | Usage in Beyla                                                                                                             |
+| ------------------------ | -------------------------------------------------------------------------------------------------------------------------- |
+| `CAP_BPF`                | enables general BPF functionality and `BPF_PROG_TYPE_SOCK_FILTER` programs                                                 |
+| `CAP_NET_RAW`            | used to create `AF_PACKET` raw sockets                                                                                     |
+| `CAP_NET_ADMIN`          | required to load `BPF_PROG_TYPE_SCHED_CLS` (tc) programs                                                                   |
+| `CAP_PERFMON`            | direct packet access and pointer arithmetic and loading `BPF_PROG_TYPE_KPROBE` programs                                    |
+| `CAP_DAC_READ_SEARCH`    | access to `/proc/self/mem` to determine kernel version                                                                     |
+| `CAP_CHECKPOINT_RESTORE` | access to symlinks in the `/proc` filesystem                                                                               |
+| `CAP_SYS_PTRACE`         | access to `/proc/pid/exe` and friends                                                                                      |
```

Comment:
I like 'and friends' but let's keep it descriptive, access to the executable modules.